### ISSUE TITLE
Rename api package to something more generic

### ISSUE THREAD

**Issue by lorenyu** [navapbc/template-application-flask#55]
# Context

current folder structure is /app/api/...
let's rename to /app/lib/... or something else that can apply to both api code and background jobs

Once we do this, we can rename `/lib/route` to `/lib/api` since the stuff in the `route` package is really stuff that's API specific (api schemas, api handlers/controller layer)

**Comment by jamesbursa** [comment]
Not sure what would be a good name here. `lib` makes me think of a shared library but this is the main server-side system code. Would we expect each project to rename it to their project name after they copy it (e.g. to `pfml`, `njui`, ...)?

**Comment by lorenyu** [comment]
@jamesbursa yeah I had that thought too, but it's not the best developer experience, since all import statements will also need to be changed.

another option could be to call it `src`, but it feels a little weird to be doing `import src.models`. any other ideas?
